public/app/plugins/datasource/loki/querybuilder/components/QueryPatternsModal.tsx

```diff
@@ -0,0 +1,125 @@
+
+import { QueryPatternsCard } from './QueryPatternsCard';
+
+type Props = {
+  isOpen: boolean;
+  query: LokiQuery;
+  queries: DataQuery[] | undefined;
+  app: CoreApp | undefined;
+  onClose: () => void;
+  onChange: (query: LokiQuery) => void;
+  onAddQuery?: (query: LokiQuery) => void;
+};
+
+export const QueryPatternsModal = (props: Props) => {
+  const { isOpen, onClose, onChange, onAddQuery, query, queries, app } = props;
+  const [openTabs, setOpenTabs] = useState<string[]>([]);
+  const [selectedPatternName, setSelectedPatternName] = useState<string | null>(null);
+
+  const styles = useStyles2(getStyles);
+  const hasNewQueryOption = !!onAddQuery;
+  const hasPreviousQuery = useMemo(
+    () => buildVisualQueryFromString(query.expr).query.operations.length > 0,
+    [query.expr]
+  );
+
+  const onPatternSelect = (pattern: LokiQueryPattern, selectAsNewQuery = false) => {
+    const visualQuery = buildVisualQueryFromString(selectAsNewQuery ? '' : query.expr);
+    reportInteraction('grafana_loki_query_patterns_selected', {
+      version: 'v2',
+      app: app ?? '',
+      editorMode: query.editorMode,
+      selectedPattern: pattern.name,
+      preSelectedOperationsCount: visualQuery.query.operations.length,
+      preSelectedLabelsCount: visualQuery.query.labels.length,
+      createNewQuery: hasNewQueryOption && selectAsNewQuery,
+    });
+
+    visualQuery.query.operations = pattern.operations;
+    if (hasNewQueryOption && selectAsNewQuery) {
+      onAddQuery({
+        ...query,
+        refId: getNextRefIdChar(queries ?? [query]),
+        expr: lokiQueryModeller.renderQuery(visualQuery.query),
+      });
+    } else {
+      onChange({
+        ...query,
+        expr: lokiQueryModeller.renderQuery(visualQuery.query),
+      });
+    }
+    setSelectedPatternName(null);
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} title="Kick start your query" onDismiss={onClose}>
+      <div className={styles.spacing}>
+        Kick start your query by selecting one of these queries. You can then continue to complete your query.
+      </div>
+      {Object.values(LokiQueryPatternType).map((patternType) => {
+        return (
+          <Collapse
+            key={patternType}
+            label={`${capitalize(patternType)} query starters`}
+            isOpen={openTabs.includes(patternType)}
+            collapsible={true}
+            onToggle={() =>
+              setOpenTabs((tabs) =>
+                // close tab if it's already open, otherwise open it
+                tabs.includes(patternType) ? tabs.filter((t) => t !== patternType) : [...tabs, patternType]
+              )
+            }
+          >
+            <div className={styles.cardsContainer}>
+              {lokiQueryModeller
+                .getQueryPatterns()
+                .filter((pattern) => pattern.type === patternType)
+                .map((pattern) => (
+                  <QueryPatternsCard
```

Comment:
```suggestion
                  <QueryPatternsCard
```
I think the `Card` part in the component name is leaking implementation details that are not relevant for consumers of this component. One day that component might have the visual representation of a Card, later it can be something else. `QueryPattern` should be enough in general.

Just something to consider.

Reply:
Yes, thanks for pointing it out! I absolutely agree. Updating this.